### PR TITLE
[WB-8733] add executor tests

### PR DESCRIPTION
**Pull request by kptkin**
Fixes WB-8733
Fixes #NNNN

Description
-----------
What does the PR do?
Adds tests for user use cases as described in Jira ticket.

Testing
-------
How was this PR tested?
These are tests

Checklist
-------
- Name PR "[WB-NNNN][WB-MMMM] Add support for..." similar to entries in CHANGELOG.md
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

File: functional_tests/mp/20-2-thread-pool-executor.yea [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  - :run_40[history][0][i]: 40
+  - :run_40[history][1][i]: 41
+  - :run_40[history][2][i]: 42
+
+  - :run_2[exitcode]: 0
+  - :run_12[exitcode]: 0
+  - :run_17[exitcode]: 0
+  - :run_40[exitcode]: 0
+
```
Comment (kptkin) [inline functional_tests/mp/20-2-thread-pool-executor.yea:56]:
```suggestion

```

File: functional_tests/mp/20-1-process-pool-executor.yea [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+plugin:
+  - wandb
+
+assert:
+  - :wandb:runs_len: 1
+  - :wandb:runs[0][history][0][a]: 0
+  - :wandb:runs[0][history][1][a]: 1
+  - :wandb:runs[0][history][2][a]: 2
+  - :history_len: 3
+  - :wandb:runs[0][exitcode]: 0
```
Comment (kptkin) [inline functional_tests/mp/20-1-process-pool-executor.yea:24]:
```suggestion
  - :wandb:runs[0][exitcode]: 0

```

File: functional_tests/mp/20-2-thread-pool-executor.yea (hunk 1 above)
Comment (kptkin) [inline functional_tests/mp/20-2-thread-pool-executor.yea:56]:
```suggestion

```

File: functional_tests/mp/20-2-thread-pool-executor.yea [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  - :run_17[history][2][i]: 19
+
+  - :run_40[history][0][i]: 40
+  - :run_40[history][1][i]: 41
+  - :run_40[history][2][i]: 42
+
+  - :run_2[exitcode]: 0
+  - :run_12[exitcode]: 0
+  - :run_17[exitcode]: 0
+  - :run_40[exitcode]: 0
```
Comment (kptkin) [inline functional_tests/mp/20-2-thread-pool-executor.yea:55]:
```suggestion
  - :run_40[exitcode]: 0

```

**Comment by dmitryduev** [comment]
I approves lol 😂

**Comment by kptkin** [comment]
> I approves lol 😂

haha... assumed it's ok since this pr only includes tests

**Pull request merged**